### PR TITLE
feat: redisPrefix

### PR DESCRIPTION
**Pull request by monwolf**
This should implement this FR https://github.com/renovatebot/renovate/issues/26394

### CODE REVIEW

File: docs/usage/self-hosted-configuration.md [hunk 1]
```diff
@@ -784,6 +784,10 @@ For TLS/SSL-enabled connections, use rediss prefix
 
 Example URL structure: `rediss://[[username]:[password]]@localhost:6379/0`.
 
+## redisPrefix
+
+If this value is set then Renovate will add this string to the name of all Redis cache entries used in Renovate. It's only used if `redisUrl` is configured.
```
Comment (rarkins) [inline docs/usage/self-hosted-configuration.md:789]:
```suggestion
If this value is set then Renovate will prepend this string to the name of all Redis cache entries used in Renovate.
It's only used if `redisUrl` is configured.
```

File: lib/config/options/index.ts [hunk 2]
```diff
@@ -334,6 +334,14 @@ const options: RenovateOptions[] = [
     type: 'string',
     globalOnly: true,
   },
+  {
+    name: 'redisPrefix',
+    description: 'If set, this will be the prefix used in all cache entries.',
```
Comment (rarkins) [inline lib/config/options/index.ts:339]:
```suggestion
    description: 'Key prefix for redis cache entries.',
```

File: lib/util/cache/package/redis.ts [hunk 3]
```diff
@@ -5,9 +5,10 @@ import { logger } from '../../../logger';
 import { compress, decompress } from '../../compress';
 
 let client: ReturnType<typeof createClient> | undefined;
+let rprefix: string | undefined;
 
 function getKey(namespace: string, key: string): string {
-  return `${namespace}-${key}`;
+  return `${rprefix}:${namespace}-${key}`;
```
Comment (rarkins) [inline lib/util/cache/package/redis.ts:11]:
```suggestion
  return `${rprefix}${namespace}-${key}`;
```

If you need a colon then add it as part of redisPrefix

File: lib/config/options/index.ts [hunk 4]
```diff
@@ -334,6 +334,14 @@ const options: RenovateOptions[] = [
...
+    stage: 'global',
+    type: 'string',
+    default: '',
```
Comment (rarkins) [inline lib/config/options/index.ts:342]:
```suggestion
```

File: lib/util/cache/package/redis.ts [hunk 5]
```diff
@@ -80,10 +81,14 @@ export async function set(
...
 
-export async function init(url: string): Promise<void> {
+export async function init(
+  url: string,
+  prefix: string | undefined,
+): Promise<void> {
   if (!url) {
     return;
   }
+  rprefix = prefix;
```
Comment (rarkins) [inline lib/util/cache/package/redis.ts:91]:
```suggestion
  rprefix = prefix ?? '';
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — DISMISSED** [review]

**Comment by monwolf** [comment]
@rarkins is there anything extra I have to change in order to be merged?

File: lib/util/cache/package/redis.ts [hunk 6]
```diff
@@ -5,9 +5,10 @@ import { logger } from '../../../logger';
 import { compress, decompress } from '../../compress';
 
 let client: ReturnType<typeof createClient> | undefined;
+let rprefix: string | undefined;
```
Comment (zharinov) [inline lib/util/cache/package/redis.ts:8]: Just `prefix` is also ok

**Review by zharinov — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**